### PR TITLE
For #11903: Fix private custom tabs

### PR DESCRIPTION
**Pull request by NotWoods**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt [hunk 1]
```diff
@@ -43,7 +44,15 @@ class IntentReceiverActivity : Activity() {
...
-        val processor = getIntentProcessors().firstOrNull { it.process(intent) }
+        val private = settings().openLinksInAPrivateTab
+        intent.putExtra(PRIVATE_BROWSING_MODE, private)
+        if (private) {
+            components.analytics.metrics.track(Event.OpenedLink(Event.OpenedLink.Mode.PRIVATE))
+        } else {
+            components.analytics.metrics.track(Event.OpenedLink(Event.OpenedLink.Mode.NORMAL))
+        }
+
+        val processor = getIntentProcessors(private).firstOrNull { it.process(intent) }
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt:55]: So AFAICT this `getIntentProcessors` will also process `externalAppIntentProcessors` and others. Before even though it wasn't working (lol) it was attempting to just add this extra on the custom tab and intent processors. Will there be any bad side effects from adding this extra to other intent entry points?

Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt:55]: There shouldn't be side effects when we add extra intents. If we want to be super safe we can set the boolean on both `this.intent` (before) and the home activity intent (now).

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt:55]: Sorry that was a confusing comment. I meant as an example - now the TWA/PWA externalAppIntentProcessors will also be attaching `PRIVATE_BROWSING_MODE`, right? Does that mean PWAs will also be attempting to present / be in PBM?

Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt:55]: Honestly we wanted it to work that way in the first place, so if they do present in PBM that fixes another bug :)

We can do another pass in the future to re-evaulate what data is getting passed in intents like this.

**Pull request merged**